### PR TITLE
Chore: Panel injects content-script

### PR DESCRIPTION
**Pull request by johnemau**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

## Short description of the change(s)

Refactor injection of the content-script from the background script to the panel, giving the panel control over what code injected; this also reduces the number of files included in the bundle and loaded by the extension.

As part of this change the build is now done in two passes; first to create the content-script bundle, then to create the extension bundle which raw-loads the content-script bundle.

Bundle for testing:
* [extension-browser-pr-1801.zip](https://github.com/webhintio/hint/files/2814725/extension-browser-pr-1801.zip)

Co-authored-by: Tony Ross <antross@gmail.com>

### CODE REVIEW

File: packages/extension-browser/webpack.config.js [hunk 1]
```diff
@@ -58,3 +53,23 @@ module.exports = {
     },
     resolve: { alias: { url$: path.resolve(__dirname, 'dist/src/shims/url.js') } }
 };
+
+const contentScriptConfig = merge.smart(baseConfig, {
```
Comment (molant) [inline packages/extension-browser/webpack.config.js]: Do we really need `merge.smart`? From what I'm reading in the [docs](https://www.npmjs.com/package/webpack-merge#smart-merging) it's more for the `loaders` stuff.
For our use case, isn't the following enough?

```ts
const contentScriptConfig = Object.assign({ }, baseConfig, {
    entry: { webhint: './dist/src/content-script/webhint.js' },
    output: { path: path.resolve(__dirname, 'dist') }
});
```

Comment (johnemau) [reply packages/extension-browser/webpack.config.js]: Correct, its future proofing if the configs ever get conflicting loaders. Should I remove for now?

Comment (molant) [reply packages/extension-browser/webpack.config.js]: If you don't mind yes. Unless we have an specific example already in mind of something we want to do relatively soon then let's remove it from here and package.json.

Thanks!

**Comment by molant** [comment]
@johnemau can you rebase also with master? There are some conflicts. Thanks!

**Comment by molant** [comment]
@antross @johnemau tests seem to fail in all platforms can you please take a look?

**Comment by antross** [comment]
@molant This should be good to go unless you have additional feedback.

**Pull request merged**